spec/lib/ddq-backend-mysql.spec.js

```diff
@@ -0,0 +1,382 @@
+                    callback(null);
+                });
+                instance.sendMessage("Example Message", "Example Topic", () => {
+                    console.log("Callback was called.");
+                });
+                expect(console.log).toHaveBeenCalledWith("Message was sent successfully.");
+                expect(console.log).toHaveBeenCalledWith("Callback was called.");
+            });
+            it("throws an error", () => {
+                instance.config.createMessageCycleLimitMs = 0;
+                expect(() => {
+                    instance.sendMessage("Example Message", "Example Topic", (err) => {
+                        throw err;
+                    });
+                }).toThrow(Error("Could not send message."));
+            });
+        });
+        describe("writeRecord()", () => {
+            it("emits an error", () => {
+                instance.connection.query.andCallFake((query, option, callback) => {
+                    callback({
+                        code: "ER_DUP_ENTRY"
+                    });
+                });
+                instance.sendMessage("Example Message", "Example Topic", () => {});
+                expect(instance.emit).toHaveBeenCalledWith("error", {
+                    code: "ER_DUP_ENTRY"
+                });
+            });
+            it("logs if a record does not exist", () => {
+                instance.connection.query.andCallFake((query, option, callback) => {
+                    if (instance.connection.query.callCount === 2) {
+                        callback(null, {
+                            affectedRows: 123
+                        });
+                    } else {
+                        callback({
+                            code: "ER_DUP_ENTRY"
+                        });
+                    }
+                });
+                instance.sendMessage("Example Message", "Example Topic", () => {});
+                expect(console.error).toHaveBeenCalledWith("Could not update record because it does not exist.");
+            });
+            it("calls the callback on success", () => {
+                instance.connection.query.andCallFake((query, option, callback) => {
+                    if (instance.connection.query.callCount === 2) {
+                        callback(null, {});
+                    } else {
+                        callback({
+                            code: "ER_DUP_ENTRY"
+                        });
+                    }
+                });
+                instance.sendMessage("Example Message", "Example Topic", () => {
+                    console.log("Callback was called.");
+                });
+                expect(console.log).toHaveBeenCalledWith("Callback was called.");
+            });
+        });
+    });
+    describe("wrapped message functions:", () => {
+        beforeEach(() => {
+            spyOn(console, "error");
+            spyOn(instance, "emit").andCallThrough();
```

Comment:
If instance.emit didn't call through, none of the wrapped message functions would be available for testing.